### PR TITLE
Added step by step quickstart to create project and access API keys i…

### PR DESCRIPTION
**Pull request by Mikerniker**
@pdeziel Hello, I tried adding the steps from the video tutorial. Will wait for feedback and if I need to make other changes. 
Thank you.

### Scope of changes

Added a step by step guide on creating a project and accessing API keys on the platform based on the video tutorials.

### Type of change

- [ ] new feature
- [ ] bug fix
- [X ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

- [ ] Check if the documentation aligns with internal processes 
- [ ] Check for typos or other errors to links 
- [ ] Check for accuracy of information provided

### Definition of Done

- [ ] I have manually tested the change running it locally (having rebuilt all containers) or via unit tests 
- [ ] I have added unit and/or integration tests that cover my changes
- [ ] I have added new test fixtures as needed to support added tests
- [ ] I have updated the dependencies list if necessary (including updating yarn.lock and/or go.sum)
- [ ] I have recompiled and included new protocol buffers to reflect changes I made if necessary
- [ X] Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [ ] I have notified the reviewer via Shortcut or Slack that this is ready for review
- [ ] Front-end: Checked sm, md, lg screen resolutions for effective responsiveness
- [ ] Backend-end: Documented service configuration changes or created related devops stories

### Reviewer(s) checklist

- [ ] Front-end: I've reviewed the Figma design and confirmed that changes match the spec.
- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: docs/content/getting-started/_index.md [hunk 1]
```diff
@@ -7,6 +7,40 @@ description: "Getting Started with Ensign"
 
 The first step is to get an Ensign API key by visiting [the sign-up page](https://rotational.app/register). Similar to getting a developer API key for [Youtube](https://developers.google.com/youtube/v3/getting-started) or [Data.gov](https://api.data.gov/docs/api-key/), you will need an API key to use Ensign and to follow along with the rest of this Quickstart guide.
 
+
+**Steps to Create Your Project and Access Your API Key**
```
Comment (pdeziel) [inline docs/content/getting-started/_index.md:11]:
```suggestion
```

File: docs/content/getting-started/_index.md [hunk 2]
```diff
@@ -7,6 +7,40 @@ description: "Getting Started with Ensign"
...
+
+**Step 1. Create a New Ensign Project**
+- After opening an account, click the `Create` button under `Set Up a New Project`.
```
Comment (pdeziel) [inline docs/content/getting-started/_index.md:14]:
```suggestion
- After creating an account, click the `Create` button under `Set Up a New Project`.
```

File: docs/content/getting-started/_index.md [hunk 3]
```diff
@@ -7,6 +7,40 @@ description: "Getting Started with Ensign"
...
+- Enter a `Project Name` (required) and provide an optional description, then click `Create Project` (You will be redirected to the Projects dashboard).
```
Comment (pdeziel) [inline docs/content/getting-started/_index.md:15]:
```suggestion
- Enter a `Project Name` (required) and provide an optional description, then click `Create Project`. You will be redirected to the Projects dashboard.
```

File: docs/content/getting-started/_index.md [hunk 4]
```diff
@@ -7,6 +7,40 @@ description: "Getting Started with Ensign"
...
+- Watch the video tutorial at [Ensign Project](https://www.youtube.com/watch?v=VskNgAVMORQ)
+
+**Step 2. Create Ensign Topics**
+- On the Project dashboard, click on your Project Name.
+- Go to the `Design Your Data Flows: Set up Your Topics` section to create topics.
```
Comment (pdeziel) [inline docs/content/getting-started/_index.md:20]:
```suggestion
```

File: docs/content/getting-started/_index.md [hunk 5]
```diff
@@ -7,6 +7,40 @@ description: "Getting Started with Ensign"
...
+**Step 1. Create a New Ensign Project**
+- After opening an account, click the `Create` button under `Set Up a New Project`.
+- Enter a `Project Name` (required) and provide an optional description, then click `Create Project` (You will be redirected to the Projects dashboard).
+- Watch the video tutorial at [Ensign Project](https://www.youtube.com/watch?v=VskNgAVMORQ)
+
+**Step 2. Create Ensign Topics**
+- On the Project dashboard, click on your Project Name.
+- Go to the `Design Your Data Flows: Set up Your Topics` section to create topics. 
+- Click `New Topic` and enter a Topic Name (required), then click `Create Topic`. 
+- Watch the video tutorial at [Ensign Topics](https://www.youtube.com/watch?v=1XuVPl_Ki4U)
```
Comment (pdeziel) [inline docs/content/getting-started/_index.md:22]:
```suggestion
- For more details, watch the [Topic Creation Tutorial](https://www.youtube.com/watch?v=1XuVPl_Ki4U)
```

File: docs/content/getting-started/_index.md [hunk 6]
```diff
@@ -7,6 +7,40 @@ description: "Getting Started with Ensign"
...
+> **Tips:**
+> - Pick a name for each topic that encodes information about the data that's stored in that topic.
```
Comment (pdeziel) [inline docs/content/getting-started/_index.md:24]:
```suggestion
> - Pick a name for each topic that encodes information about the data you intend to store in that topic.
```

File: docs/content/getting-started/_index.md [hunk 7]
```diff
@@ -7,6 +7,40 @@ description: "Getting Started with Ensign"
...
+> - You can also include information about the data format (e.g. flights-json or hotels-html). Read more at [Naming Topics]({{< ref "/getting-started/topics" >}})
```
Comment (pdeziel) [inline docs/content/getting-started/_index.md:25]:
```suggestion
> - You can also include information about the data format (e.g. flights-json or hotels-html). Read more [here]({{< ref "/getting-started/topics" >}})
```

File: docs/content/getting-started/_index.md [hunk 8]
```diff
@@ -7,6 +7,40 @@ description: "Getting Started with Ensign"
...
+> - Topics can help serve as a data contract helping upstream publishers remember how to serialize the data and downstream subscribers how to parse it.
+
+**Step 3. Create Ensign API Keys**
+- Go to the `Permission Your Data Flows: Generate API Keys` section and click `New Key`.
```
Comment (pdeziel) [inline docs/content/getting-started/_index.md:29]:
```suggestion
- Click the `New Key` button.
```

File: docs/content/getting-started/_index.md [hunk 9]
```diff
@@ -7,6 +7,40 @@ description: "Getting Started with Ensign"
...
+- A `Generate API Key for project` pop-up will open, add a `Key Name,` and choose `Permissions` (Full Access or Custom Access), then click `Generate API Key`.
```
Comment (pdeziel) [inline docs/content/getting-started/_index.md:30]:
```suggestion
- Create a name for the key and select its permissions (Full Access or Custom Access), then click `Generate API Key`.
```

File: docs/content/getting-started/_index.md [hunk 10]
```diff
@@ -7,6 +7,40 @@ description: "Getting Started with Ensign"
...
+- From the `Your API Key` popup, copy both the `Client ID` and the `Client Secret` to open a client connection to Ensign. Read More at [Ensign API Keys](#ensign-api-keys)
```
Comment (pdeziel) [inline docs/content/getting-started/_index.md:31]:
```suggestion
- Copy or download the `Client ID` and the `Client Secret`, you will need both to connect to Ensign. Read more [here](#ensign-api-keys) 
```

File: docs/content/getting-started/_index.md [hunk 11]
```diff
@@ -7,6 +7,40 @@ description: "Getting Started with Ensign"
...
+> **Tips:**
+> - Pick a name for each topic that encodes information about the data that's stored in that topic.
+> - You can also include information about the data format (e.g. flights-json or hotels-html). Read more at [Naming Topics]({{< ref "/getting-started/topics" >}})
+> - Topics can help serve as a data contract helping upstream publishers remember how to serialize the data and downstream subscribers how to parse it.
+
+**Step 3. Create Ensign API Keys**
+- Go to the `Permission Your Data Flows: Generate API Keys` section and click `New Key`.
+- A `Generate API Key for project` pop-up will open, add a `Key Name,` and choose `Permissions` (Full Access or Custom Access), then click `Generate API Key`.
+- From the `Your API Key` popup, copy both the `Client ID` and the `Client Secret` to open a client connection to Ensign. Read More at [Ensign API Keys](#ensign-api-keys) 
+- Watch the video tutorial at [Creating Ensign API Keys](https://www.youtube.com/watch?v=KMejrUIouMw)
```
Comment (pdeziel) [inline docs/content/getting-started/_index.md:32]:
```suggestion
- For more details, watch the [API Key Creation Tutorial](https://www.youtube.com/watch?v=KMejrUIouMw)
```

File: docs/content/getting-started/_index.md [hunk 12]
```diff
@@ -7,6 +7,40 @@ description: "Getting Started with Ensign"
...
+> **Important:**
+> - Remember to download your keys and save this file.
```
Comment (pdeziel) [inline docs/content/getting-started/_index.md:34]:
```suggestion
```

File: docs/content/getting-started/_index.md [hunk 13]
```diff
@@ -7,6 +7,40 @@ description: "Getting Started with Ensign"
...
+> - Add the `Client ID` and `Client Secret` to your environment variables, or you can put the file into your Git repository via gitignore. Read More at [Authentication](#authentication)
+> - Do not share your keys with anyone, and never commit them to a public GitHub repository.
```
Comment (pdeziel) [inline docs/content/getting-started/_index.md:36]:
```suggestion
> - Do not share your keys with anyone, and never commit them to a public GitHub repository. Read more about best practices for API keys [here](#authentication).
```

File: docs/content/getting-started/_index.md [hunk 14]
```diff
@@ -7,6 +7,40 @@ description: "Getting Started with Ensign"
...
+> - Topics can help serve as a data contract helping upstream publishers remember how to serialize the data and downstream subscribers how to parse it.
+
+**Step 3. Create Ensign API Keys**
+- Go to the `Permission Your Data Flows: Generate API Keys` section and click `New Key`.
+- A `Generate API Key for project` pop-up will open, add a `Key Name,` and choose `Permissions` (Full Access or Custom Access), then click `Generate API Key`.
+- From the `Your API Key` popup, copy both the `Client ID` and the `Client Secret` to open a client connection to Ensign. Read More at [Ensign API Keys](#ensign-api-keys) 
+- Watch the video tutorial at [Creating Ensign API Keys](https://www.youtube.com/watch?v=KMejrUIouMw)
+> **Important:**
+> - Remember to download your keys and save this file.
+> - Add the `Client ID` and `Client Secret` to your environment variables, or you can put the file into your Git repository via gitignore. Read More at [Authentication](#authentication)
```
Comment (pdeziel) [inline docs/content/getting-started/_index.md:35]:
```suggestion
```

File: docs/content/getting-started/_index.md [hunk 15]
```diff
@@ -7,6 +7,40 @@ description: "Getting Started with Ensign"
...
+> - Do not share your keys with anyone, and never commit them to a public GitHub repository.
+> - If your keys get lost or compromised, don't worry, you can revoke them and create new ones.
+> - API keys are defined at the project level, and everyone collaborating on the datasets will need their own API keys.
```
Comment (pdeziel) [inline docs/content/getting-started/_index.md:38]:
```suggestion
> - API keys grant access to all topics in the project.
```

File: docs/content/getting-started/_index.md [hunk 16]
```diff
@@ -7,6 +7,40 @@ description: "Getting Started with Ensign"
...
+> - Once you add teammates to your project, they can create their own API keys using the same technique.
```
Comment (pdeziel) [inline docs/content/getting-started/_index.md:39]:
```suggestion
> - Team members with the `Member`, `Owner`, or `Admin` role can create API keys, an `Observer` can not create API keys.
```

File: docs/content/getting-started/_index.md [hunk 17]
```diff
@@ -7,6 +7,40 @@ description: "Getting Started with Ensign"
...
+> - New API keys enable users or applications to publish new data to the project topics, or to subscribe to those topics.
```
Comment (pdeziel) [inline docs/content/getting-started/_index.md:40]:
```suggestion
```

**Review by pdeziel — CHANGES_REQUESTED** [review]
Awesome PR, I really like the tips and important sections, they look really good in the docs.

I did have several suggestions to streamline the wording and such but overall this is a great addition.

**Comment by Mikerniker** [comment]
Okay, thank you @pdeziel , I tried to commit your suggestions, I'm not sure if I did this correctly, but let me know if I missed something or if I still need to make other changes. Thank you for your help
.

**Review by pdeziel — APPROVED** [review]
Looks great, thanks for making the updates! I'll merge the PR once the checks pass.

**Comment by Mikerniker** [comment]
Ok, thanks @pdeziel !

**Comment by Mikerniker** [comment]
Okay, let me try to study  and figure out how to set up a GPG key and if I get lost, I'll maybe go for the second option. Will get back on this when or if I am able to set it up. Thanks!

**Comment by Mikerniker** [comment]
Hi @pdeziel I have set up the GPG signature and I'm currently trying to figure out how to open a new PR. --> when I click open a new PR I keep getting redirected to this page... am I supposed to click "Close with comment" button here first? Sorry, I'm still new at making PRs...

**Comment by pdeziel** [comment]
> Hi @pdeziel I have set up the GPG signature and I'm currently trying to figure out how to open a new PR. --> when I click open a new PR I keep getting redirected to this page... am I supposed to click "Close with comment" button here first? Sorry, I'm still new at making PRs...

@Mikerniker The way I would do it is create a new branch on your fork and open a PR from that one with the signed commit. Unfortunately if you add signed commits on this branch it still won't work, because the original commit you added here yesterday was not signed and it wants all the commits to be signed.

**Comment by Mikerniker** [comment]
Ok got it, will try that now, thanks!

**Pull request closed without merging**